x/gamm/keeper/migrate.go

```diff
@@ -245,7 +307,10 @@ func (k Keeper) ReplaceMigrationRecords(ctx sdk.Context, records []types.Balance
 
 	migrationInfo.BalancerToConcentratedPoolLinks = records
 
-	k.OverwriteMigrationRecords(ctx, migrationInfo)
+	err = k.OverwriteMigrationRecordsAndRedirectDistrRecords(ctx, migrationInfo)
```

Comment:
Just occurred to me – it doesn't seem like we check if `migrationInfo` refers to links that actually exist in state. If I'm not mistaken, this could trigger a panic when `deleteMigrationKeys` right?

Reply:
It seems appropriate for us to panic if gov attempts create a link to non-existent pools

Reply:
Hmm right but we should be catching this in validation logic so the proposal in invalidated without ever triggering the panic right?

Might be a nonissue so feel free to disregard if you're confident this is fine

Reply:
Yeah I agree that this should be checked in validation logic (I thought it was, but I could be wrong)

Reply:
Should have clarified – what prompted the comment was the realization that we don't check this in validation logic (likely since it's general validation that also covers the case where records don't already exist in state):
https://github.com/osmosis-labs/osmosis/blob/eb54eaff14d84bfda1c6febbd93377e3e2c57e9d/x/gamm/keeper/migrate.go#L149

Reply:
@AlpinYukseloglu do you think if we just remove deleteMigrationKeys from Overwrite that this should be good?

Reply:
This makes much more sense, thanks for the clarification